nodes.go

```diff
@@ -225,6 +226,96 @@ func (n *Node) GetMetadata() (string, error) {
 	return md.String(), nil
 }
 
+type updateMetadata struct {
+	Name        *string `json:"name,omitempty"`
+	Description *string `json:"description,omitempty"`
+}
+
+// Updates metadata fields name or description of Node n. If a given field is nil,
```

Comment:
Go comments for external function should be in the style `// FuncName ...` so this should be `// UpdateMetadata changes name or ...`

https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

Though I see you removed this in the next changeset ;-)